private/rules/v2_lock_file.bzl

```diff
@@ -43,11 +43,20 @@ def _compute_lock_file_hash(lock_file_contents):
     return hash(repr(to_hash))
 
 def _to_m2_path(unpacked):
-    path = "{group}/{artifact}/{version}/{artifact}-{version}".format(
-        artifact = unpacked["artifactId"],
-        group = unpacked["groupId"].replace(".", "/"),
-        version = unpacked["version"],
-    )
+    version = unpacked["version"]
+    if "dirVersion" in unpacked:
+        path = "{group}/{artifact}/{dirVersion}/{artifact}-{version}".format(
+            artifact = unpacked["artifactId"],
+            group = unpacked["groupId"].replace(".", "/"),
+            version = unpacked["version"],
+            dirVersion = unpacked["dirVersion"],
+        )
+    else:
+        path = "{group}/{artifact}/{version}/{artifact}-{version}".format(
+            artifact = unpacked["artifactId"],
+            group = unpacked["groupId"].replace(".", "/"),
+            version = unpacked["version"],
+        )
```

Comment:
This could have less redundancy if you doubled-down on the notion of `dirVersion` in both cases:
```starlark
if "dirVersion" in unpacked:
  dir_version = unpacked["dir_version"]
else:
  dir_version = unpacked["version"]

path = "{group}/{artifact}/{dir_version}/{artifact}-{version}".format(
    artifact = unpacked["artifactId"],
    dir_version = dir_version,
    group = unpacked["groupId"].replace(".", "/"),
    version = unpacked["version"],
)
```

(Also note style nit about using `snake_case` for variables.)

Reply:
Done.